File: Java/lab9/src/main/java/org/example/Pc.java
package org.example;

import java.util.ArrayList;
import java.util.List;


public class Pc implements Component{
    List<Component> components=new ArrayList<>();
    public void addComponent(Component component){
        components.add(component);
    }
    public void removeComponent(Component component){
        if(!components.isEmpty()){
            components.remove(component);
        }else {
            System.out.println("no component to repair");
        }
    }
    @Override
    public void repair() {
        System.out.println("\nWe repair component");
        for (Component component : components) {
            component.repair();
        }
    }

}

File: Python/lab9/Pc.py
from Component import Component

class Pc(Component):
    def __init__(self):
        self.components = []

    def addComponent(self, component: Component):
        self.components.append(component)

    def removeComponent(self, component: Component):
        self.components.remove(component)

    def repair(self):
        print("\nWe repair")
        for component in self.components:
            component.repair()
